In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv("../Dataset/processed/zomato_analytics.csv")

df.head()

,name,location,cuisines,rest_type,online_order,book_table,rate,votes,approx_cost,listed_in_type,listed_in_city
0,Jalsa,Banashankari,"North Indian, Mughlai, Chinese",Casual Dining,Yes,Yes,4.10,775,800.0,Buffet,Banashankari\n
1,Spice Elephant,Banashankari,"Chinese, North Indian, Thai",Casual Dining,Yes,No,4.10,787,800.0,Buffet,Banashankari\n
2,San Churro Cafe,Banashankari,"Cafe, Mexican, Italian","Cafe, Casual Dining",Yes,No,3.80,918,800.0,Buffet,Banashankari\n
3,Addhuri Udupi Bhojana,Banashankari,"South Indian, North Indian",Quick Bites,No,No,3.70,88,300.0,Buffet,Banashankari\n
4,Grand Village,Basavanagudi,"North Indian, Rajasthani",Casual Dining,No,No,3.80,166,600.0,Buffet,Banashankari\n


In [2]:
df.shape

(51717, 11)

In [3]:
df.isnull().sum()

name                0
location           21
cuisines           45
rest_type         227
online_order        0
book_table          0
rate                0
votes               0
approx_cost       346
listed_in_type      0
listed_in_city      0
dtype: int64

In [4]:
(df.isnull().sum()/len(df))*100

name              0.000000
location          0.040606
cuisines          0.087012
rest_type         0.438927
online_order      0.000000
book_table        0.000000
rate              0.000000
votes             0.000000
approx_cost       0.669026
listed_in_type    0.000000
listed_in_city    0.000000
dtype: float64

In [6]:
df.duplicated().sum()

np.int64(108)

In [7]:
df = df.drop_duplicates()

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 51609 entries, 0 to 51716
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   name            51609 non-null  object 
 1   location        51588 non-null  object 
 2   cuisines        51564 non-null  object 
 3   rest_type       51382 non-null  object 
 4   online_order    51609 non-null  object 
 5   book_table      51609 non-null  object 
 6   rate            51609 non-null  object 
 7   votes           51609 non-null  int64  
 8   approx_cost     51265 non-null  float64
 9   listed_in_type  51609 non-null  object 
 10  listed_in_city  51609 non-null  object 
dtypes: float64(1), int64(1), object(9)
memory usage: 4.7+ MB


In [9]:
df["approx_cost"] = pd.to_numeric(df["approx_cost"], errors="coerce")

In [10]:
df["name"] = df["name"].str.strip()

In [11]:
df["location"] = df["location"].str.strip()

In [12]:
df["cuisines"] = df["cuisines"].str.strip()

In [13]:
df["rest_type"] = df["rest_type"].str.strip()

In [14]:
df["rate"].isnull().sum()

np.int64(0)

In [15]:
df["votes"].describe()

count    51609.000000
mean       283.283361
std        803.282771
min          0.000000
25%          7.000000
50%         41.000000
75%        198.000000
max      16832.000000
Name: votes, dtype: float64

In [16]:
df["approx_cost"].describe()

count    51265.000000
mean       555.170682
std        438.587661
min         40.000000
25%        300.000000
50%        400.000000
75%        650.000000
max       6000.000000
Name: approx_cost, dtype: float64

In [17]:
df.info()

df.describe()

df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
Index: 51609 entries, 0 to 51716
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   name            51609 non-null  object 
 1   location        51588 non-null  object 
 2   cuisines        51564 non-null  object 
 3   rest_type       51382 non-null  object 
 4   online_order    51609 non-null  object 
 5   book_table      51609 non-null  object 
 6   rate            51609 non-null  object 
 7   votes           51609 non-null  int64  
 8   approx_cost     51265 non-null  float64
 9   listed_in_type  51609 non-null  object 
 10  listed_in_city  51609 non-null  object 
dtypes: float64(1), int64(1), object(9)
memory usage: 4.7+ MB


name                0
location           21
cuisines           45
rest_type         227
online_order        0
book_table          0
rate                0
votes               0
approx_cost       344
listed_in_type      0
listed_in_city      0
dtype: int64

In [18]:
df.to_csv("../Dataset/processed/zomato_final.csv", index=False)

# Initial Observations

- The dataset loaded successfully from the SQL-processed CSV.
- The dataset contains **51,717 restaurant records** and **11 analysis-ready columns**.
- Primary data cleaning has already been completed in SQL, including:
  - Removal of invalid rating values (`NEW` and `-`)
  - Standardization of the `rate` column
  - Removal of commas from the `approx_cost` column
- Most columns appear to have the correct structure for analysis.
- Some columns may still contain missing values that require validation.
- Text columns may contain leading or trailing whitespace that should be standardized.
- Duplicate records, if any, need to be identified and handled.
- Numeric columns such as `rate`, `votes`, and `approx_cost` will be validated to ensure they have the correct data types.
- The cleaned dataset will be verified before proceeding to feature engineering, visualization, and business analysis.

## Objective

The objective of this notebook is to validate the SQL-cleaned dataset, perform any remaining preprocessing, ensure data quality, and prepare a final analysis-ready dataset for Python analysis and Power BI dashboard development.

# Key Findings

- Verified dataset dimensions and data types.
- Confirmed SQL cleaning was successfully applied.
- Standardized text fields for consistency.
- Checked and handled missing values where required.
- Removed duplicate records (if present).
- Validated numeric columns for further analysis.
- Exported the final cleaned dataset (`zomato_final.csv`) for feature engineering and visualization.